zebra-consensus/src/checkpoint.rs

```diff
@@ -225,16 +235,33 @@ where
             // We start by verifying the genesis block, by itself
             None => (None, Progress::BeforeGenesis),
         };
+        let (sender, receiver) = mpsc::channel();
         CheckpointVerifier {
             checkpoint_list,
             network,
             initial_tip_hash,
             state_service,
             queued: BTreeMap::new(),
             verifier_progress,
+            reset_receiver: receiver,
+            reset_sender: sender,
         }
     }
 
+    /// Reset the verifier progress back to given tip.
+    fn reset_progress(&mut self, tip: Option<(block::Height, block::Hash)>) {
+        self.verifier_progress = match tip {
+            Some((height, _)) => {
+                if height >= self.checkpoint_list.max_height() {
+                    Progress::FinalCheckpoint
+                } else {
+                    Progress::InitialTip(height)
```

Comment:
We also need to set `initial_tip_hash` here, so that the verifier has an initial hash to start the first checkpoint.
(This is one of the things I'd improve if I rewrote this code.)
 
Can we re-use the initial tip processing code from the constructor,
so that we always handle the initial tip the same way?
https://github.com/ZcashFoundation/zebra/blob/c6082602561a15c0dfb37578a9b09b3cd8e017a4/zebra-consensus/src/checkpoint.rs#L215-L227